### PR TITLE
DRAFT when we do run it, its gives error

### PR DESCRIPTION
**Pull request by yehudalev**

### CODE REVIEW

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java [hunk 1]
```diff
@@ -1,24 +1,287 @@
...
+      Iterator<CSVRecord> iter = parser.iterator();
+
+      //begin this is for next use when we parse from the file, we are listing the lenght of each message 
+      int offset = 0;
+      List<Integer> offsetlist = new ArrayList<Integer>();
+
+      while (iter.hasNext()) {
+        CSVRecord record = iter.next();
+        StopTime.Builder stoptime = StopTime.newBuilder();
+        stoptime.setTripId(record.get(0)).setArrivalTime(record.get(1)).setDepartureTime(record.get(2))
```
Comment (val-fed) [inline projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: Please use access to data by headers instead of the access by index.

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java [hunk 2]
```diff
@@ -1,24 +1,287 @@
...
+            .setStopId(record.get(3)).setStopSequence(record.get(4)).setPickupType(record.get(5))
```
Comment (val-fed) [inline projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: Let's use a separate line for each action in the chain. It will be more readable. I mean:
```
 stoptime
            .setTripId(record.get(0))
            .setArrivalTime(record.get(1))
            ....
            .setShapeDistTraveled(record.get(7))
            .build()
            .writeTo(output);
```

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java [hunk 3]
```diff
@@ -1,24 +1,287 @@
...
+            .setDropOffType(record.get(6)).setShapeDistTraveled(record.get(7)).build().writeTo(output);
+        //for next use when we parse from the file
+        offset = stoptime.build().toByteArray().length;
+        offsetlist.add(offset);
+      }
+    } catch (IOException e) {
```
Comment (val-fed) [inline projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: The catch block is empty. You can throw `new RuntimeException(e)`.

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java [hunk 4]
```diff
@@ -1,24 +1,287 @@
...
+    }
+    return allShapesBuilder;
+  }
+
+  public static void getStopTime(Path csvPath, String path) {
+    try {
+      //where we serialize the messages
+      FileOutputStream output = new FileOutputStream(path);
```
Comment (val-fed) [inline projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: Looks like you don't close the stream. You can use try-with-resources construction.

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java (hunk 3 above)
Comment (yehudalev) [reply projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: When I do that, it give me error, where exactry I need to pu the `new RunTimeException(e)`

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java (hunk 4 above)
Comment (yehudalev) [reply projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: Can you rephrase, please?

File: projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java (hunk 3 above)
Comment (val-fed) [reply projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java]: It can look like:
```
try {
    // some code
} catch (IOException e) {
  throw new RuntimeException(e);
}
```
